dcos_test_utils/package.py

```diff
@@ -93,3 +93,218 @@ def list_packages(self):
         """
         self._update_headers('list')
         return self._post('/list', {})
+
+
+class Repository(Cosmos):
+    def __init__(self, default_url, session=None):
+        super().__init__(default_url, session=session)
+
+    def add(self, name: str, uri: str, index: int = None) -> dict:
+        """Add a package repository. If `index` is 0 it will be added to
+        the top of the repository list.
+
+        :param name: Repository name
+        :type name: str
+        :param uri: Repository URI
+        :type uri: str
+        :param index: Position in repository list starting at 0
+            (lower is higher priority)
+        :type index: int
+        :return: JSON response
+        :rtype: dict
+        """
+        params = {
+            'uri':  uri,
+            'name': name,
+        }
+
+        if index is not None:
+            idx_type = type(index)
+            if idx_type != int:
+                raise TypeError('index of type {} is not supported '
+                                '- must be int'.format(idx_type))
+            params['index'] = index
+
+        self._update_headers('repository.add')
+        r = self._post('/add', params)
+        return r.json()
+
+    def delete(self, name: str = None, uri: str = None) -> dict:
+        """Delete the package repository with given name.
+
+        :param name: Repository name
+        :type name: str
+        :param uri: Repository URI
+        :type uri: str
+        :return: JSON response
+        :rtype: dict
+        """
+        params = {}
```

Comment:
Is a param required? Or can someone do `r.delete()` with no params? If required, I'd raise an exception if neither is set like you do above with the int type check.

Reply:
Not sure - thanks for the question.